components/director/internal/domain/application/resolver.go

```diff
@@ -336,6 +343,20 @@ func (r *Resolver) DeleteApplication(ctx context.Context, id string) (*graphql.A
 		return nil, err
 	}
 
+	auths, err := r.sysAuthSvc.ListForObject(ctx, model.ApplicationReference, app.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, auth := range auths {
```

Comment:
I wonder if this logic shouldn't be moved to service. Best to consult.

Reply:
Yes, please move the for loop to oauth20Svc and just call single method